go_lib/hooks/set_cr_statuses/hook.go

```diff
@@ -0,0 +1,46 @@
+/*
+Copyright 2023 Flant JSC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package set_cr_statuses
+
+import (
+	"fmt"
+	"os"
+	"time"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+func getTimeStamp() string {
+	curTime := time.Now()
+	if timeStr, ok := os.LookupEnv("TEST_CONDITIONS_CALC_NOW_TIME"); ok {
+		curTime, _ = time.Parse(time.RFC3339, timeStr)
+	}
+	return curTime.Format(time.RFC3339)
+}
+
+var SetProcessedStatus = func(obj *unstructured.Unstructured) (*unstructured.Unstructured, error) {
```

Comment:
```suggestion
func SetProcessedStatus(obj *unstructured.Unstructured) (*unstructured.Unstructured, error) {
```

This is a pitfall because a user can change the value of this variable. It is better to make it a package method because it is immutable and exported at the same time.